systemservices/marketplace/src/tasks/prepareCreateServiceOffer.ts

```diff
@@ -1,16 +1,28 @@
 import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
 import { Marketplace } from "../contracts/Marketplace"
 import { toUnit, asciiToHex, CreateTransaction } from "../contracts/utils";
+import BigNumber from "bignumber.js";
+import { getService } from "../contracts/service";
 
 export default (
   marketplace: Marketplace,
   createTransaction: CreateTransaction
 ) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
   try {
+    // check inputs
+    const duration = new BigNumber(inputs.duration)
+    if (duration.isNegative() || duration.isZero()) throw new Error('Duration cannot be negative or equal to zero')
+
+    // check service
+    const service = await getService(marketplace, inputs.sid)
+    if (service.owner.toLowerCase() !== inputs.from.toLowerCase()) throw new Error(`Service's owner is different that the specified 'from'`)
+    if (service.versions.length === 0) throw new Error('Cannot create an offer on a service with 0 version')
```

Comment:
a service must be published with at least 1 version at the beginning, why do we have this check here?

Reply:
Because the smart contract can be called without using the service and doesn't prevent this case.

Reply:
how about updating the smart contact to check this?

Reply:
The current smart contract structure totally allow this.
Let's think about this for the next version of the smart contract.